### PR TITLE
Migrated Dockerfile to use alpine linux

### PR DESCRIPTION
**Pull request by Luzifer**
@jbenet PTAL

TODO:
- [x] Check checksum for downloaded asset

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 6f50c700ba0883f134881e1af939e8369b540d98
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23)

---
This message was auto-generated by https://gitcop.com

**Comment by jbenet** [comment]
this LGTM. rerunning tests

**Comment by jbenet** [comment]
thanks @Luzifer !

**Comment by jbenet** [comment]
also cc @lgierth for CR

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 6f50c700ba0883f134881e1af939e8369b540d98
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

 * Commit: 9f34551663f98624814518f2416a4be5b699194e
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23)

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 6f50c700ba0883f134881e1af939e8369b540d98
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

 * Commit: 9f34551663f98624814518f2416a4be5b699194e
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

 * Commit: 6fb9fee74923e68f4beb7d2c2cf8ca229c448bc7
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23)

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a245a27c17cc90c7fc26147aa65b075acf7223e7
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23)

---
This message was auto-generated by https://gitcop.com

**Comment by Luzifer** [comment]
@jbenet PTAL again. Now with checksum checking

**Comment by lgierth** [comment]
I'd prefer if we build from source, for a number of reasons:

- we don't need the shacheck script
- we can build images from local changes or feature branches (important, imo)
- we can strip down the size a bit more by using the musl libc

**Comment by jbenet** [comment]
But then we need the go compiler, which isn't small.




+1 for building from branches.



The hash check should prob still happen (if we can get go to be reproducible).





—

**Comment by lgierth** [comment]
Anyhow, let's go with this for now :+1: 

My comments are not a blocker regarding that blog posting.

**Comment by jbenet** [comment]
@Luzifer gitcop is complaining because you have
```
License: MIT Signed-off-by: Knut Ahlers <knut@ahlers.me>
```
it expects:
```
License: MIT
Signed-off-by: Knut Ahlers <knut@ahlers.me>
```
maybe use: https://github.com/ipfs/community/blob/master/dev/tools/hooks/setup_commit_msg_hook.sh

(doesn't matter for this, i'll merge it)

File: bin/container_shacheck [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -xe
+
+VERSION=$1
+FILENAME=$2
+
+ONLINE_SHA=$( curl "https://gobuilder.me/api/v1/github.com/ipfs/go-ipfs/cmd/ipfs/signed-hashes/${VERSION}" 2>/dev/null | grep -A 4 ${FILENAME} | grep sha1 | awk '{ print $3 }' )
+
+echo "Checking SHA1: ${ONLINE_SHA} == $(sha1sum ${FILENAME} | awk '{print $1}')"
+
+echo "${ONLINE_SHA}  ${FILENAME}" | sha1sum -cw
```
Comment (jbenet) [inline bin/container_shacheck]: ideally we should use the `multihash` tool, but that can be an improvement later.

**Comment by jbenet** [comment]
thanks @Luzifer @lgierth

**Comment by RX14** [comment]
Would it not be better to make the Dockerfile build from current source rather than the latest gobuilder, because then with the docker auto build you get actual latest, and I presume the docker auto builds are faster than the gobuilder builds.

**Comment by balupton** [comment]
this seems undone by looking at the latest master - any solutions for using go-ipfs on alpine linux?

**Comment by lgierth** [comment]
@balupton you'll have to build go-ipfs yourself in an Alpine container (or another way that makes sure it's linked against musl libc). We generally don't have containers or prebuilt binaries linked with musl, but we'd like to in the future.

**Pull request merged**